core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -94,6 +98,86 @@ public final void setPort(final int port) {
         }
     }
 
+    /**
+     * Gets the name or literal IP address of the host that the client is configured to connect to.
+     *
+     * @return The host name.
+     */
+    public final String getProxyHost() {
+        return proxyHost;
+    }
+
+    /**
+     * Sets the name or literal IP address of the proxy host that the client should use to connect to the an AMQP host
+     *
+     * @param proxyHost The host name or IP address of the proxy
+     * @throws NullPointerException if proxyHost is {@code null}.
+     */
+    public final void setProxyHost(final String proxyHost) {
+        this.proxyHost = Objects.requireNonNull(proxyHost);
+    }
+
+    /**
+     * Gets the TCP port of the proxy server that this client is configured to use.
+     *
+     * @return The port number.
+     */
+    public final int getProxyPort() {
+        return proxyPort;
+    }
+
+    /**
+     * Gets the username of the proxy server the client is configured with.
+     *
+     * @return the username
+     */
+    public String getProxyUsername() {
+        return proxyUsername;
+    }
+
+    /**
+     * sets the username to be used for authentication with the proxy server.
+     *
+     * @param proxyUsername username to be used for authentication with the proxy server
+     * @throws NullPointerException if proxyUsername is {@code null}.
+     */
+    public void setProxyUsername(String proxyUsername) {
+        this.proxyUsername = Objects.requireNonNull(proxyUsername);
+    }
+
+    /**
+     * Gets the password of the proxy user to be used in authentication.
+     *
+     * @return password in clear text.
+     */
+    public String getProxyPassword() {
+        return proxyPassword;
+    }
+
+    /**
+     * Sets the password of the proxy user to be used in authentication.
+     *
+     * @param proxyPassword password in clear text
+     * @throws NullPointerException if proxyUsername is {@code null}.
+     */
+    public void setProxyPassword(String proxyPassword) {
+        this.proxyPassword = proxyPassword;
+    }
+
+    /**
+     * Sets the TCP port of the proxy server that this client should use.
+     *
+     * @param proxyPort The port number of the proxy
+     * @throws IllegalArgumentException if port is 0.
+     */
+    public final void setProxyPort(final int proxyPort) {
```

Comment:
can you please move this up to the corresponding getter?

Reply:
done.